pkg/settings/setting_test.go

```diff
@@ -76,3 +91,36 @@ func TestSystemFeatureChartRefreshSecondsDefault(t *testing.T) {
 	}
 
 }
+
+func TestGetInt(t *testing.T) {
```

Comment:
Isn't it an issue that we're mutating/accessing the `settings` global map concurrently without locks?

For example, if `TestGetInt` runs at the same time as `Test_ChartInstallation` (pkg/controllers/dashboard/systemcharts/controller_test.go).

In fact, my question still applies even without `t.Parallel()` because tests in other modules could also be creating new settings, etc.

Reply:
`t.Parallel` runs tests concurrently only with **other** parallel tests. Also, those are accesses of different entries in the map, so this is safe? It's bad that the map is shared, but that's a fundamental package design flaw, which, ideally, we should fix sometime later during package cleanup and refactoring.

Reply:
> t.Parallel runs tests concurrently only with other parallel tests

So from the docs I see that `t.Parallel()` runs tests concurrently within this module only. But tests in separate modules can be run in parallel. Here's what the docs says:

```
$ go help testflag
...
	-parallel n
...
The 'go test' command may run tests for different packages
	   in parallel as well, according to the setting of the -p flag
	   (see 'go help build').
```


> so this is safe?

It's not safe, here's an example you can run with `go test -v -race` and will output failures:

```go
package main

import (
	"testing"
)

var (
	Settings map[string]string
)

func init() {
	Settings = make(map[string]string, 1)
}

func TestFoo(t *testing.T) {
	t.Parallel()
	Settings["foo"] = "foo"
}

func TestBar(t *testing.T) {
	t.Parallel()
	Settings["bar"] = "bar"
}
```

For some reason I'm unable to make it fail if I split those two tests in separate packages.. So either they are **not** run in parallel or the race detector doesn't work the way I understand it..

> ideally, we should fix sometime later during package cleanup and refactoring

Has this issue already been brought up and is planned?